sci-mathematics/gmp_ecm/gmp_ecm-7.0.4.recipe

```diff
@@ -0,0 +1,70 @@
+SUMMARY="Elliptic Curve Method for Integer Factorization"
+DESCRIPTION="C implementation of the Elliptic Curve Method for factorizing integers"
+HOMEPAGE="http://ecm.gforge.inria.fr/"
+COPYRIGHT="2001-2016 GMP-ECM Authors"
+LICENSE="
+	GNU GPL v3
+	GNU LGPL v3
+	"
+REVISION="1"
+SOURCE_URI="https://gforge.inria.fr/frs/download.php/file/36224/ecm-$portVersion.tar.gz"
+CHECKSUM_SHA256="0cf7b3eee8462cc6f98b418b47630e1eb6b3f4f8c3fc1fb005b08e2a1811ba43"
+SOURCE_DIR="ecm-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	gmp_ecm$secondaryArchSuffix = $portVersion
+	cmd:ecm$secondaryArchSuffix = $portVersion
+	lib:libecm$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	gmp_ecm${secondaryArchSuffix}_devel = $portVersion
+	devel:libecm$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	haiku$secondaryArchSuffix
+	lib:libecm$secondaryArchSuffix = $portVersion
```

Comment:
Two little mistakes:
```diff
 REQUIRES_devel="
-	haiku$secondaryArchSuffix
+	haiku${secondaryArchSuffix}_devel
-	lib:libecm$secondaryArchSuffix = $portVersion
+	gmp_ecm$secondaryArchSuffix == $portVersion
	"
```

Reply:
This is an exception.
Although the general rule is to declare the dependencies on libs with `lib:` lines, the dependency of a package's devel sub-package on its own base package should always be specified using the name of its own base package.

In other words, if foo_devel needs bar_devel (i.e. another package providing e.g. libbar) then we add a `devel:libbar$secondaryArchSuffix` in foo's REQUIRES_devel (and a `lib:libbar$secondaryArchSuffix` in foo's REQUIRES).
**But** in foo's REQUIRES_devel we *must not* put `lib:foo$secondaryArchSuffix` *but* `foo$secondaryArchSuffix`, even if the devel package does not need any `cmd:` item from its base package.
This is to make sure the devel package will depend on its base package.

Hmm, speaking about the "base" keyword, I forgot to add it in me previous comment. (I'll fix this in a second.)